validator/gen/gen.go

```diff
@@ -0,0 +1,16 @@
+package main
```

Comment:
Do you mind moving the generation code to `gen/gen`? Here we use a single file for code generation to make it simpler.

Reply:
Fixed